crates/bevy_ui/src/widget/text.rs

```diff
@@ -47,9 +50,9 @@ pub fn text_system(
     mut font_atlas_set_storage: ResMut<Assets<FontAtlasSet>>,
     mut text_pipeline: ResMut<DefaultTextPipeline>,
     mut text_queries: ParamSet<(
-        Query<Entity, Or<(Changed<Text>, Changed<Style>)>>,
-        Query<Entity, (With<Text>, With<Style>)>,
-        Query<(&Text, &Style, &mut CalculatedSize)>,
+        Query<Entity, Or<(Changed<Text>, FlexboxLayoutChanged)>>,
+        Query<Entity, (With<Text>, With<FlexboxLayout>)>,
+        Query<(&Text, FlexboxLayoutQuery, &mut CalculatedSize)>,
```

Comment:
I really like how the queries are now more precise and the system no longer triggers runs when unrelated properties changes

Reply:
Agreed. I expect it'll be possible to refine this behavior further, but I wanted to start with a very direct port for this PR.